mlflow/tracing/client.py

```diff
@@ -531,7 +535,16 @@ def _get_artifact_repo_for_trace(self, trace_info: TraceInfo):
         artifact_uri = add_databricks_profile_info_to_artifact_uri(artifact_uri, self.tracking_uri)
         return get_artifact_repository(artifact_uri)
 
-    def _download_trace_data(self, trace_info: TraceInfo) -> TraceData:
+    def _download_trace_data(self, trace_info) -> TraceData:
```

Comment:
```suggestion
    def _download_trace_data(self, trace_info: Union[TraceInfo, TraceInfoV3]) -> TraceData:
```

Same as above, let's not remove type hint - and ideally update upstream to use `TraceInfoV3` only.